sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -269,4 +279,441 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          assessmentDate: addDays(30, programYear.startDate),
+        },
+      },
+    );
+    const firstAssessmentDate =
+      currentApplication.currentAssessment.assessmentDate;
+    // The start date for the first SFAS and SFAS part time application record is set to the date before the first assessment date of the current application.
+    const firstLegacyApplicationStartDate = addDays(-10, firstAssessmentDate);
+    const firstLegacyApplicationEndDate = addDays(30, firstAssessmentDate);
+    // The start date for the second SFAS and SFAS part time application record is set to the date after the end date of the first SFAS application.
+    const secondLegacyApplicationStartDate = addDays(
+      10,
+      firstLegacyApplicationEndDate,
+    );
+    const secondLegacyApplicationEndDate = addDays(
+      40,
+      firstLegacyApplicationEndDate,
+    );
+    const secondAssessment = createFakeStudentAssessment(
+      {
+        auditUser: currentApplication.student.user,
+        application: currentApplication,
+        offering: currentApplication.currentAssessment.offering,
+      },
+      {
+        initialValue: {
+          assessmentWorkflowId: "some fake id",
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+        },
+      },
+    );
+    currentApplication.currentAssessment = secondAssessment;
+    await db.dataSource.getRepository(Application).save(currentApplication);
+
+    // SFAS Individual.
+    const sfasIndividual = await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: student.user.lastName,
+        birthDate: student.birthDate,
+        sin: student.sinValidation.sin,
+      },
+    });
+    // First SFAS application with the start date before the first assessment date of the current application.
+    const firstFakeSFASApplication = createFakeSFASApplication(
+      { individual: sfasIndividual },
+      {
+        initialValues: {
+          startDate: getISODateOnlyString(firstLegacyApplicationStartDate),
+          endDate: getISODateOnlyString(firstLegacyApplicationEndDate),
+          csgdAward: 9,
+          csgpAward: 10,
+          sbsdAward: 12,
+          bcagAward: 13,
+        },
+      },
+    );
+    // Second SFAS application with the start date after the end date of the first SFAS application.
+    const secondFakeSFASApplication = createFakeSFASApplication(
+      { individual: sfasIndividual },
+      {
+        initialValues: {
+          startDate: getISODateOnlyString(secondLegacyApplicationStartDate),
+          endDate: getISODateOnlyString(secondLegacyApplicationEndDate),
+          csgdAward: 9,
+          csgpAward: 10,
+          sbsdAward: 12,
+          bcagAward: 13,
+        },
+      },
+    );
+    await db.dataSource
+      .getRepository(SFASApplication)
```

Comment:
If `SFASApplication` does not exists in the `dataSource` it must be added and then used here.